burner-wallet-plugin/tokenbridge-bw-exchange/src/utils/abis/Mediator.ts

```diff
@@ -34,5 +34,45 @@ export default [
     payable: false,
     stateMutability: 'view',
     type: 'function'
+  },
+  {
+    constant: false,
+    inputs: [
+      {
+        name: '',
+        type: 'address'
+      },
+      {
+        name: '',
+        type: 'uint256'
+      }
+    ],
+    name: 'relayTokens',
+    outputs: [],
+    payable: false,
+    stateMutability: 'nonpayable',
+    type: 'function'
+  },
+  {
+    anonymous: false,
+    inputs: [
+      {
+        indexed: true,
+        name: '',
+        type: 'address'
+      },
+      {
+        indexed: true,
+        name: '',
+        type: 'address'
+      },
+      {
+        indexed: false,
+        name: '',
+        type: 'uint256'
+      }
+    ],
+    name: 'Transfer',
```

Comment:
The mediator should not emit this event. Could you clarify why it is here?

Reply:
When tokens are sent through the bridge using `relayTokens`, the method is called on the bridge contract instance: https://github.com/poanetwork/tokenbridge/blob/5ee9dea0c44cf8f2b055333886cc79f0ffc6251c/burner-wallet-plugin/tokenbridge-bw-exchange/src/burner-wallet/assets/BridgeableERC20Asset.ts#L37-L39
This call results in Transfer event being emitted from the token contract, however, when web3 parses receipt logs, it knows only information about events from the called contract. So I added this ABI here in order to further access parsed Transfer event at https://github.com/poanetwork/tokenbridge/blob/5ee9dea0c44cf8f2b055333886cc79f0ffc6251c/burner-wallet-plugin/tokenbridge-bw-exchange/src/burner-wallet/assets/BridgeableERC20Asset.ts#L43

Reply:
Yeah, I understood this. This is a sort of shortcut but could lead to a confusion in the future. Is this more correct to get the transaction hash and use the token contract to get the Transfer event by using the transaction receipt? Yes, it will add more code but this is more correct to reflect which contract which event emitted.

Reply:
Ok, agree. Since we don't use parsed event and the only thing we need is its log index, we can find a suitable log by comparing its topic with the existing `TRANSFER_TOPIC` constant.